packages/jest-runtime/src/index.ts

```diff
@@ -1270,11 +1272,28 @@ class Runtime {
           this,
           from.filename,
         )) as LocalModuleRequire;
-    moduleRequire.cache = Object.create(null);
     moduleRequire.extensions = Object.create(null);
     moduleRequire.requireActual = this.requireActual.bind(this, from.filename);
     moduleRequire.requireMock = this.requireMock.bind(this, from.filename);
     moduleRequire.resolve = resolve;
+    moduleRequire.cache = ((): RequireCache => {
+      const moduleRegistry = this._moduleRegistry;
+      const notPermittedMethod = () => {
+        console.warn('`require.cache` modification is not permitted');
+        return true;
+      };
+      return new Proxy(Object.create(null), {
```

Comment:
```suggestion
      return new Proxy<RequireCache>(Object.create(null), {
```

Types the functions correctly